figure out the eslint rules for the tests directory

Per @vladikoff's https://github.com/mozilla/fxa-js-client/pull/158#issuecomment-116774697
According to @TDA in https://github.com/mozilla/fxa-js-client/pull/158#discussion_r32364977, linting the tests/* directory leads to ~7117 errors/warnings.

@vladikoff It looks like the majority of those 7000+ errors were coming from the code in [./tests/addons/sinon.js](https://github.com/mozilla/fxa-js-client/blob/438b76fd13361bfb4d8eb599bdce8ccd51bdda85/tests/addons/sinon.js).

After excluding that file, the number drops to a far more manageable 130+/-. And if we temporarily remove the formatting rules (`semi`, `quotes`, `indent`, etc), the number drops to around 9 issues:
```
$ grunt eslint:tests
Running "eslint:tests" (eslint) task

tests/lib/account.js
  347:35  error  "account" is not defined  no-undef
  353:29  error  "account" is not defined  no-undef

tests/lib/certificateSign.js
  15:10  warning  mail is defined but never used        no-unused-vars
  18:10  warning  ErrorMocks is defined but never used  no-unused-vars

tests/lib/errors.js
  15:10  warning  mail is defined but never used          no-unused-vars
  17:10  warning  RequestMocks is defined but never used  no-unused-vars

tests/lib/recoveryEmail.js
  18:10  warning  ErrorMocks is defined but never used  no-unused-vars

tests/lib/verifyCode.js
  13:10  warning  accountHelper is defined but never used  no-unused-vars
  18:10  warning  ErrorMocks is defined but never used     no-unused-vars

✖ 9 problems (2 errors, 7 warnings)

Warning: Task "eslint:tests" failed. Use --force to continue.

Aborted due to warnings.
```

Ok this one looks super-legit, account really isnt there
tests/lib/account.js
  347:35  error  "account" is not defined  no-undef
  353:29  error  "account" is not defined  no-undef
@pdehaan 
@vladikoff

--- a/.jscsrc
+++ b/.jscsrc
@@ -1,5 +1,5 @@
 {
-    "disallowKeywords": ["with", "eval"],
+    "disallowKeywords": ["eval"],
     "disallowKeywordsOnNewLine": ["else"],
     "requireSpaceBeforeBinaryOperators": ["?", "-", "/", "*", "=", "==", "===", "!=", "!==", ">", ">=", "<", "<="],
     "disallowMultipleLineStrings": true,
@@ -9,12 +9,10 @@
     "disallowSpaceBeforePostfixUnaryOperators": ["++", "--"],
     "maximumLineLength": 420,
     "requireCapitalizedConstructors": true,
-    "requireCurlyBraces": ["if", "else", "for", "while", "do"],
     "requireLineFeedAtFileEnd": true,
     "requireSpaceAfterKeywords": ["if", "else", "for", "while", "do", "switch", "return"],
     "validateIndentation": 2,
     "validateLineBreaks": "LF",
-    "validateQuoteMarks": true,
     "validateJSDoc": {
         "checkParamNames": true,
         "checkRedundantParams": true,

--- a/tasks/eslint.js
+++ b/tasks/eslint.js
@@ -4,15 +4,22 @@
 
 module.exports = function (grunt) {
   grunt.config('eslint', {
-    options: {
-      eslintrc: '.eslintrc'
-    },
     app: {
+      eslintrc: '.eslintrc',
       src: ['Gruntfile.js', 'tasks/*.js', 'config/**/*.js', 'node/**/*.js']
     },
     client: {
       options: {eslintrc: 'client/.eslintrc'},
       src: ['client/*.js', 'client/lib/**/*']
+    },
+    test: {
+      options: {
+        eslintrc: 'tests/.eslintrc'
+      },
+      src: [
+        'tests/**/*.js',
+        '!tests/addons/sinon.js'
+      ]
     }
   });
 };

--- a/tasks/jscs.js
+++ b/tasks/jscs.js
@@ -8,11 +8,12 @@ module.exports = function (grunt) {
   grunt.config('jscs', {
     src: [
       '**/*.js',
+      'tests/**/*.js',
       '!node_modules/**',
       '!build/**',
       '!components/**',
       '!docs/**',
-      '!tests/**'
+      '!tests/addons/sinon.js'
     ],
     options: {
       config: '.jscsrc'

--- a/tests/.eslintrc
+++ b/tests/.eslintrc
@@ -1,0 +1,8 @@
+{
+  "env": {
+    "mocha": true
+  },
+  "rules": {
+    "no-with": 0
+  }
+}

--- a/tests/addons/accountHelper.js
+++ b/tests/addons/accountHelper.js
@@ -16,8 +16,8 @@ define([
 
   AccountHelper.prototype.newVerifiedAccount = function () {
     var user = 'testHelp1' + new Date().getTime();
-    var email = user + "@restmail.net";
-    var password = "iliketurtles";
+    var email = user + '@restmail.net';
+    var password = 'iliketurtles';
     var respond = this.respond;
     var mail = this.mail;
     var client = this.client;
@@ -40,7 +40,7 @@ define([
       .then(function (emails) {
         var code = emails[0].html.match(/code=([A-Za-z0-9]+)/)[1];
 
-        return respond(client.verifyCode(uid, code),RequestMocks.verifyCode);
+        return respond(client.verifyCode(uid, code), RequestMocks.verifyCode);
       })
 
       .then(function (res) {
@@ -57,8 +57,8 @@ define([
 
   AccountHelper.prototype.newUnverifiedAccount = function () {
     var user = 'testHelp2' + new Date().getTime();
-    var email = user + "@restmail.net";
-    var password = "iliketurtles";
+    var email = user + '@restmail.net';
+    var password = 'iliketurtles';
     var respond = this.respond;
     var client = this.client;
     var result = {

--- a/tests/addons/restmail.js
+++ b/tests/addons/restmail.js
@@ -16,7 +16,7 @@ define([
   Restmail.prototype.wait = function (user, number) {
     var self = this;
 
-    if (!number) number = 1;
+    if (!number) number = 1; //eslint-disable-line curly
     console.log('Waiting for email...');
 
     return self.request.send('/mail/' + user, 'GET')

--- a/tests/addons/sinonResponder.js
+++ b/tests/addons/sinonResponder.js
@@ -21,7 +21,7 @@ define([
         });
 
         return returnValue;
-      }
+      };
     },
     respond: function (req, mock) {
       if (typeof mock === 'undefined') {
@@ -31,5 +31,5 @@ define([
         req.respond(mock.status, mock.headers, mock.body);
       }
     }
-  }
+  };
 });

--- a/tests/intern.js
+++ b/tests/intern.js
@@ -24,28 +24,28 @@ define(['intern/lib/args'], function (args) {
     if (server === 'LATEST') {
       server = 'https://api-accounts-latest.dev.lcip.org';
     }
-    console.log("Running against " + server);
+    console.log('Running against ' + server);
   } else {
-    console.log("Running with mocks...");
+    console.log('Running with mocks...');
   }
 
   return {
-  loader: {
+    loader: {
     // Packages that should be registered with the loader in each testing environment
-    packages: [ { name: 'fxa-js-client', location: 'client' } ],
-    map: {
-      '*': {
-        sjcl: 'components/sjcl/sjcl',
-        p: 'components/p/p'
+      packages: [ { name: 'fxa-js-client', location: 'client' } ],
+      map: {
+        '*': {
+          sjcl: 'components/sjcl/sjcl',
+          p: 'components/p/p'
+        }
       }
-    }
-  },
+    },
 
-  suites: [ 'tests/all' ],
-  functionalSuites: [ ],
-  AUTH_SERVER_URL: server,
+    suites: [ 'tests/all' ],
+    functionalSuites: [ ],
+    AUTH_SERVER_URL: server,
 
-  excludeInstrumentation: /./
+    excludeInstrumentation: /./
 
   };
 

--- a/tests/intern_sauce.js
+++ b/tests/intern_sauce.js
@@ -22,13 +22,13 @@ define([
   };
 
   intern.environments = [
-    { browserName: 'firefox', version: '25' , platform: [ 'Windows 7', 'Linux' ] },
+    { browserName: 'firefox', version: '25', platform: [ 'Windows 7', 'Linux' ] },
     { browserName: 'internet explorer', version: ['9', '10'], platform: [ 'Windows 7' ] },
     { browserName: 'chrome' },
     { browserName: 'safari' }
   ];
 
-  console.log("SAUCE", intern.proxyUrl);
+  console.log('SAUCE', intern.proxyUrl);
 
   return intern;
 });

--- a/tests/lib/account.js
+++ b/tests/lib/account.js
@@ -10,12 +10,12 @@ define([
 
   with (tdd) {
     suite('account', function () {
-       var accountHelper;
-       var respond;
-       var mail;
-       var client;
-       var RequestMocks;
-       var ErrorMocks;
+      var accountHelper;
+      var respond;
+      var mail;
+      var client;
+      var RequestMocks;
+      var ErrorMocks;
 
       beforeEach(function () {
         var env = new Environment();
@@ -36,13 +36,13 @@ define([
             email = account.input.email;
             password = account.input.password;
 
-            return respond(client.accountDestroy(email, password), RequestMocks.accountDestroy)
+            return respond(client.accountDestroy(email, password), RequestMocks.accountDestroy);
           })
           .then(
             function(res) {
               assert.ok(res, 'got response');
 
-              return respond(client.signIn(email, password), ErrorMocks.accountDoesNotExist)
+              return respond(client.signIn(email, password), ErrorMocks.accountDoesNotExist);
             }
           ).then(
             function () {
@@ -59,7 +59,7 @@ define([
 
         return accountHelper.newVerifiedAccount()
           .then(function (account) {
-            return respond(client.accountKeys(account.signIn.keyFetchToken, account.signIn.unwrapBKey), RequestMocks.accountKeys)
+            return respond(client.accountKeys(account.signIn.keyFetchToken, account.signIn.unwrapBKey), RequestMocks.accountKeys);
           })
           .then(
             function(keys) {
@@ -78,13 +78,13 @@ define([
             account = acc;
             var incorrectCaseEmail = account.input.email.charAt(0).toUpperCase() + account.input.email.slice(1);
 
-            return respond(client.accountDestroy(incorrectCaseEmail, account.input.password), RequestMocks.accountDestroy)
+            return respond(client.accountDestroy(incorrectCaseEmail, account.input.password), RequestMocks.accountDestroy);
           })
           .then(
           function(res) {
             assert.ok(res);
 
-            return respond(client.signIn(account.input.email, account.input.password), ErrorMocks.accountDoesNotExist)
+            return respond(client.signIn(account.input.email, account.input.password), ErrorMocks.accountDoesNotExist);
           }
         ).then(
           function () {
@@ -106,7 +106,7 @@ define([
             account = acc;
             var incorrectCaseEmail = account.input.email.charAt(0).toUpperCase() + account.input.email.slice(1);
 
-            return respond(client.accountDestroy(incorrectCaseEmail, account.input.password, {skipCaseError: true}), ErrorMocks.incorrectEmailCase)
+            return respond(client.accountDestroy(incorrectCaseEmail, account.input.password, {skipCaseError: true}), ErrorMocks.incorrectEmailCase);
           })
           .then(
             function () {
@@ -133,26 +133,26 @@ define([
         return respond(client.signUp(email, password), RequestMocks.signUp)
           .then(function (result) {
             uid = result.uid;
-            assert.ok(uid, "uid is returned");
+            assert.ok(uid, 'uid is returned');
 
             return respond(client.passwordForgotSendCode(email), RequestMocks.passwordForgotSendCode);
           })
           .then(function (result) {
             passwordForgotToken = result.passwordForgotToken;
-            assert.ok(passwordForgotToken, "passwordForgotToken is returned");
+            assert.ok(passwordForgotToken, 'passwordForgotToken is returned');
 
             return respond(mail.wait(user, 2), RequestMocks.resetMail);
           })
           .then(function (emails) {
             var code = emails[1].html.match(/code=([A-Za-z0-9]+)/)[1];
-            assert.ok(code, "code is returned: " + code);
+            assert.ok(code, 'code is returned: ' + code);
 
             return respond(client.passwordForgotVerifyCode(code, passwordForgotToken), RequestMocks.passwordForgotVerifyCode);
           })
           .then(function (result) {
             accountResetToken = result.accountResetToken;
             var newPassword = 'newturles';
-            assert.ok(accountResetToken, "accountResetToken is returned");
+            assert.ok(accountResetToken, 'accountResetToken is returned');
 
             return respond(client.accountReset(email, newPassword, accountResetToken), RequestMocks.accountReset);
           })
@@ -161,7 +161,7 @@ define([
               assert.isNotNull(result);
             },
             assert.notOk
-          )
+          );
       });
 
       test('#passwordForgotSendCode with service, redirectTo, and resume', function () {
@@ -176,7 +176,7 @@ define([
           .then(function (acc) {
             account = acc;
 
-            return respond(client.passwordForgotSendCode(account.input.email, opts), RequestMocks.passwordForgotSendCode)
+            return respond(client.passwordForgotSendCode(account.input.email, opts), RequestMocks.passwordForgotSendCode);
           })
           .then(function (result) {
             assert.ok(result.passwordForgotToken);
@@ -197,7 +197,7 @@ define([
             assert.equal(service[1], 'sync', 'service is returned');
             assert.equal(redirectTo[1], 'https', 'redirectTo is returned');
             assert.equal(resume[1], 'resumejwt', 'resume is returned');
-          })
+          });
       });
 
       test('#passwordForgotResendCode with service, redirectTo, and resume', function () {
@@ -213,7 +213,7 @@ define([
           .then(function (acc) {
             account = acc;
 
-            return respond(client.passwordForgotSendCode(account.input.email, opts), RequestMocks.passwordForgotSendCode)
+            return respond(client.passwordForgotSendCode(account.input.email, opts), RequestMocks.passwordForgotSendCode);
           })
           .then(function (result) {
             assert.ok(result.passwordForgotToken);
@@ -225,7 +225,7 @@ define([
             assert.ok(res);
 
             return respond(client.passwordForgotResendCode(account.input.email, passwordForgotToken, opts),
-              RequestMocks.passwordForgotResendCode)
+              RequestMocks.passwordForgotResendCode);
           })
           .then(function (result) {
             assert.ok(result.passwordForgotToken);
@@ -246,7 +246,7 @@ define([
             assert.equal(service[1], 'sync', 'service is returned');
             assert.equal(redirectTo[1], 'https', 'redirectTo is returned');
             assert.equal(resume[1], 'resumejwt', 'resume is returned');
-          })
+          });
       });
 
       test('#passwordForgotStatus', function () {
@@ -258,7 +258,7 @@ define([
           })
           .then(function (result) {
 
-            return respond(client.passwordForgotStatus(result.passwordForgotToken), RequestMocks.passwordForgotStatus)
+            return respond(client.passwordForgotStatus(result.passwordForgotToken), RequestMocks.passwordForgotStatus);
           })
           .then(
             function (result) {
@@ -266,7 +266,7 @@ define([
               assert.property(result, 'ttl');
             },
             assert.notOk
-          )
+          );
       });
 
       test('#passwordForgotStatus error with a false token', function () {
@@ -279,7 +279,7 @@ define([
           .then(function () {
             var fakeToken = 'e838790265a45f6ee1130070d57d67d9bb20953706f73af0e34b0d4d92f10000';
 
-            return respond(client.passwordForgotStatus(fakeToken), ErrorMocks.invalidAuthToken)
+            return respond(client.passwordForgotStatus(fakeToken), ErrorMocks.invalidAuthToken);
           })
           .then(
             assert.notOk,
@@ -287,7 +287,7 @@ define([
               assert.equal(err.code, 401);
               assert.equal(err.errno, 110);
             }
-          )
+          );
       });
 
       test('#accountStatus', function () {
@@ -302,7 +302,7 @@ define([
               assert.equal(res.exists, true);
             },
             assert.notOk
-          )
+          );
       });
 
       test('#accountStatus with wrong uid', function () {
@@ -313,7 +313,7 @@ define([
               assert.equal(res.exists, false);
             },
             assert.notOk
-        )
+        );
       });
 
       test('#accountStatus with no uid', function () {
@@ -344,13 +344,13 @@ define([
 
       test('#accountLock with no email', function () {
         assert.throws(function () {
-          client.accountLock(null, account.input.password);
+          client.accountLock(null, account.input.password); //eslint-disable-line no-undef
         });
       });
 
       test('#accountLock with no password', function () {
         assert.throws(function () {
-          client.accountLock(account.input.email, null);
+          client.accountLock(account.input.email, null); //eslint-disable-line no-undef
         });
       });
 
@@ -369,7 +369,7 @@ define([
             return respond(client.accountLock(account.input.email, account.input.password), RequestMocks.accountLock);
           })
           .then(function () {
-            return respond(client.accountUnlockResendCode(account.input.email, opts), RequestMocks.accountUnlockResendCode)
+            return respond(client.accountUnlockResendCode(account.input.email, opts), RequestMocks.accountUnlockResendCode);
           })
           .then(
             function (result) {
@@ -378,7 +378,7 @@ define([
               assert.equal(Object.keys(result).length, 0);
             },
             assert.notOk
-          )
+          );
       });
 
       test('#accountUnlockResendCode with no email', function () {
@@ -403,7 +403,7 @@ define([
             return respond(client.accountLock(account.input.email, account.input.password), RequestMocks.accountLock);
           })
           .then(function () {
-            return respond(client.accountUnlockResendCode(account.input.email, opts), RequestMocks.accountUnlockResendCode)
+            return respond(client.accountUnlockResendCode(account.input.email, opts), RequestMocks.accountUnlockResendCode);
           })
           .then(function () {
             return respond(mail.wait(account.input.user, 2), RequestMocks.resetMailUnlock);

--- a/tests/lib/certificateSign.js
+++ b/tests/lib/certificateSign.js
@@ -12,19 +12,15 @@ define([
     suite('certificateSign', function () {
       var accountHelper;
       var respond;
-      var mail;
       var client;
       var RequestMocks;
-      var ErrorMocks;
 
       beforeEach(function () {
         var env = new Environment();
         accountHelper = env.accountHelper;
         respond = env.respond;
-        mail = env.mail;
         client = env.client;
         RequestMocks = env.RequestMocks;
-        ErrorMocks = env.ErrorMocks;
       });
 
       test('#basic', function () {
@@ -38,7 +34,7 @@ define([
             };
             var duration = 86400000;
 
-            return respond(client.certificateSign(account.signIn.sessionToken, publicKey, duration), RequestMocks.certificateSign)
+            return respond(client.certificateSign(account.signIn.sessionToken, publicKey, duration), RequestMocks.certificateSign);
           })
           .then(
             function(res) {

--- a/tests/lib/errors.js
+++ b/tests/lib/errors.js
@@ -12,18 +12,14 @@ define([
     suite('errors', function () {
       var accountHelper;
       var respond;
-      var mail;
       var client;
-      var RequestMocks;
       var ErrorMocks;
 
       beforeEach(function () {
         var env = new Environment();
         accountHelper = env.accountHelper;
         respond = env.respond;
-        mail = env.mail;
         client = env.client;
-        RequestMocks = env.RequestMocks;
         ErrorMocks = env.ErrorMocks;
       });
 
@@ -34,7 +30,7 @@ define([
             var pk = {algorithm: 'RS', n: 'x', e: 'y'};
             var duration = 1000;
 
-            return respond(client.certificateSign(account.signIn.sessionToken, pk, duration), ErrorMocks.accountUnverified)
+            return respond(client.certificateSign(account.signIn.sessionToken, pk, duration), ErrorMocks.accountUnverified);
           })
           .then(
             function () {
@@ -51,7 +47,7 @@ define([
 
         return accountHelper.newUnverifiedAccount()
           .then(function (account) {
-            return respond(client.verifyCode(account.signUp.uid, 'eb531a64deb628b2baeaceaa8762abf0'), ErrorMocks.invalidVerification)
+            return respond(client.verifyCode(account.signUp.uid, 'eb531a64deb628b2baeaceaa8762abf0'), ErrorMocks.invalidVerification);
           })
           .then(
             function () {

--- a/tests/lib/headerLang.js
+++ b/tests/lib/headerLang.js
@@ -26,9 +26,9 @@ define([
       });
 
       test('#signUp', function () {
-        var user = "test" + new Date().getTime();
-        var email = user + "@restmail.net";
-        var password = "iliketurtles";
+        var user = 'test' + new Date().getTime();
+        var email = user + '@restmail.net';
+        var password = 'iliketurtles';
         var opts = {
           lang: 'zh-cn;'
         };
@@ -59,11 +59,11 @@ define([
           .then(function (acc) {
             account = acc;
 
-            return respond(client.passwordForgotSendCode(account.input.email, opts), RequestMocks.passwordForgotSendCode)
+            return respond(client.passwordForgotSendCode(account.input.email, opts), RequestMocks.passwordForgotSendCode);
           })
           .then(function (result) {
             passwordForgotToken = result.passwordForgotToken;
-            assert.ok(passwordForgotToken, "passwordForgotToken is returned");
+            assert.ok(passwordForgotToken, 'passwordForgotToken is returned');
 
             return respond(mail.wait(account.input.user, 2), RequestMocks.resetMailLang);
           })
@@ -73,7 +73,7 @@ define([
               assert.equal(emails[1].headers['content-language'], 'zh-CN');
             },
             assert.notOk
-          )
+          );
       });
 
       test('#recoveryEmailResendCode', function () {
@@ -86,7 +86,7 @@ define([
           .then(function (account) {
             user = account.input.user;
 
-            return respond(client.recoveryEmailResendCode(account.signIn.sessionToken, opts), RequestMocks.recoveryEmailResendCode)
+            return respond(client.recoveryEmailResendCode(account.signIn.sessionToken, opts), RequestMocks.recoveryEmailResendCode);
           })
           .then(
           function(res) {
@@ -100,7 +100,7 @@ define([
               assert.equal(emails[1].headers['content-language'], 'zh-CN');
             },
             assert.notOk
-          )
+          );
       });
 
     });

--- a/tests/lib/passwordChange.js
+++ b/tests/lib/passwordChange.js
@@ -53,7 +53,6 @@ define([
             newUnwrapBKey = sjcl.codec.hex.fromBits(newCreds.unwrapBKey);
             return respond(client.signUp(email, password), RequestMocks.signUp);
           })
-
           .then(function (result) {
             uid = result.uid;
 
@@ -64,7 +63,6 @@ define([
 
             return respond(client.verifyCode(uid, code), RequestMocks.verifyCode);
           })
-
           .then(function() {
             return respond(client.signIn(email, password, {keys: true}), RequestMocks.signInWithKeys);
           })
@@ -74,7 +72,6 @@ define([
           .then(function(keys) {
             kB = keys.kB;
           })
-
           .then(function () {
             return respond(client._passwordChangeStart(email, password), RequestMocks.passwordChangeStart);
           })
@@ -90,7 +87,7 @@ define([
           .then(function (result) {
             // currently only available for mocked requests (issue #103)
             if (requests) {
-              var req = requests[requests.length-1];
+              var req = requests[requests.length - 1];
               var args = JSON.parse(req.requestBody);
               var expectedNewWrapKB = sjcl.codec.hex.fromBits(
                 credentials.xor(sjcl.codec.hex.toBits(kB),
@@ -108,7 +105,7 @@ define([
             function (err) {
               throw err;
             }
-          )
+          );
       });
 
       test('#with incorrect case', function () {
@@ -145,7 +142,7 @@ define([
             function (err) {
               throw err;
             }
-          )
+          );
       });
 
       test('#with incorrect case with skipCaseError', function () {
@@ -197,7 +194,7 @@ define([
             return respond(client._passwordChangeStart(email, password), RequestMocks.passwordChangeStart);
           })
           .then(function (credentials) {
-            oldCreds = credentials
+            oldCreds = credentials;
 
             return respond(client._passwordChangeKeys(oldCreds), RequestMocks.accountKeys);
           })
@@ -219,7 +216,7 @@ define([
               assert.equal(error.message, 'Incorrect password', '== Password is incorrect');
               assert.equal(error.code, 400, '== Correct status code');
             }
-          )
+          );
       });
     });
   }

--- a/tests/lib/recoveryEmail.js
+++ b/tests/lib/recoveryEmail.js
@@ -15,7 +15,6 @@ define([
       var mail;
       var client;
       var RequestMocks;
-      var ErrorMocks;
 
       beforeEach(function () {
         var env = new Environment();
@@ -24,7 +23,6 @@ define([
         mail = env.mail;
         client = env.client;
         RequestMocks = env.RequestMocks;
-        ErrorMocks = env.ErrorMocks;
       });
 
       test('#recoveryEmailResendCode', function () {
@@ -34,7 +32,7 @@ define([
           .then(function (account) {
             user = account.input.user;
 
-            return respond(client.recoveryEmailResendCode(account.signIn.sessionToken), RequestMocks.recoveryEmailResendCode)
+            return respond(client.recoveryEmailResendCode(account.signIn.sessionToken), RequestMocks.recoveryEmailResendCode);
           })
           .then(
           function(res) {
@@ -46,7 +44,7 @@ define([
             function (emails) {
               // second email, the code is resent.
               var code = emails[1].html.match(/code=([A-Za-z0-9]+)/)[1];
-              assert.ok(code, "code is returned");
+              assert.ok(code, 'code is returned');
             },
             assert.notOk
           );
@@ -64,7 +62,7 @@ define([
           .then(function (account) {
             user = account.input.user;
 
-            return respond(client.recoveryEmailResendCode(account.signIn.sessionToken, opts), RequestMocks.recoveryEmailResendCode)
+            return respond(client.recoveryEmailResendCode(account.signIn.sessionToken, opts), RequestMocks.recoveryEmailResendCode);
           })
           .then(
           function(res) {

--- a/tests/lib/request.js
+++ b/tests/lib/request.js
@@ -22,7 +22,7 @@ define([
       });
 
       test('#heartbeat', function () {
-        var heartbeatRequest = env.respond(request.send("/__heartbeat__", "GET"), RequestMocks.heartbeat)
+        var heartbeatRequest = env.respond(request.send('/__heartbeat__', 'GET'), RequestMocks.heartbeat)
           .then(
             function (res) {
               assert.ok(res);
@@ -36,7 +36,7 @@ define([
       test('#error', function () {
         request = new Request('http://', env.xhr);
 
-        request.send("/", "GET")
+        request.send('/', 'GET')
           .then(
             assert.notOk,
             function () {
@@ -49,7 +49,7 @@ define([
       test('#timeout', function () {
         request = new Request('http://google.com:81', env.xhr, { timeout: 200 });
 
-        var timeoutRequest = env.respond(request.send("/", "GET"), ErrorMocks.timeout);
+        var timeoutRequest = env.respond(request.send('/', 'GET'), ErrorMocks.timeout);
 
         return timeoutRequest.then(
           assert.notOk,
@@ -63,7 +63,7 @@ define([
         request = new Request('http://example.com/', env.xhr);
 
         // Trigger an error response that's in HTML
-        var response = env.respond(request.send("/nonexistent", "GET"), ErrorMocks.badResponseFormat);
+        var response = env.respond(request.send('/nonexistent', 'GET'), ErrorMocks.badResponseFormat);
 
         return response.then(
           assert.notOk,

--- a/tests/lib/session.js
+++ b/tests/lib/session.js
@@ -30,7 +30,7 @@ define([
         return accountHelper.newVerifiedAccount()
           .then(function (account) {
 
-            return respond(client.sessionDestroy(account.signIn.sessionToken), RequestMocks.sessionDestroy)
+            return respond(client.sessionDestroy(account.signIn.sessionToken), RequestMocks.sessionDestroy);
           })
           .then(
             function(res) {
@@ -45,7 +45,7 @@ define([
         return accountHelper.newVerifiedAccount()
           .then(function (account) {
 
-            return respond(client.sessionStatus(account.signIn.sessionToken), RequestMocks.sessionStatus)
+            return respond(client.sessionStatus(account.signIn.sessionToken), RequestMocks.sessionStatus);
           })
           .then(
             function(res) {
@@ -61,7 +61,7 @@ define([
           .then(function () {
             var fakeToken = 'e838790265a45f6ee1130070d57d67d9bb20953706f73af0e34b0d4d92f10000';
 
-            return respond(client.passwordForgotStatus(fakeToken), ErrorMocks.invalidAuthToken)
+            return respond(client.passwordForgotStatus(fakeToken), ErrorMocks.invalidAuthToken);
           })
           .then(
           assert.notOk,
@@ -69,7 +69,7 @@ define([
             assert.equal(err.code, 401);
             assert.equal(err.errno, 110);
           }
-        )
+        );
       });
 
     });

--- a/tests/lib/signIn.js
+++ b/tests/lib/signIn.js
@@ -26,8 +26,8 @@ define([
       });
 
       test('#basic', function () {
-        var email = "test" + new Date().getTime() + "@restmail.net";
-        var password = "iliketurtles";
+        var email = 'test' + new Date().getTime() + '@restmail.net';
+        var password = 'iliketurtles';
 
         return respond(client.signUp(email, password), RequestMocks.signUp)
           .then(function () {
@@ -43,8 +43,8 @@ define([
       });
 
       test('#with keys', function () {
-        var email = "test" + new Date().getTime() + "@restmail.net";
-        var password = "iliketurtles";
+        var email = 'test' + new Date().getTime() + '@restmail.net';
+        var password = 'iliketurtles';
 
         return respond(client.signUp(email, password), RequestMocks.signUp)
           .then(function (res) {
@@ -61,8 +61,8 @@ define([
       });
 
       test('#with service', function () {
-        var email = "test" + new Date().getTime() + "@restmail.net";
-        var password = "iliketurtles";
+        var email = 'test' + new Date().getTime() + '@restmail.net';
+        var password = 'iliketurtles';
 
         return respond(client.signUp(email, password), RequestMocks.signUp)
           .then(function () {
@@ -71,8 +71,8 @@ define([
       });
 
       test('#with reason', function () {
-        var email = "test" + new Date().getTime() + "@restmail.net";
-        var password = "iliketurtles";
+        var email = 'test' + new Date().getTime() + '@restmail.net';
+        var password = 'iliketurtles';
 
         return respond(client.signUp(email, password), RequestMocks.signUp)
           .then(function () {

--- a/tests/lib/signUp.js
+++ b/tests/lib/signUp.js
@@ -29,8 +29,8 @@ define([
       });
 
       test('#basic', function () {
-        var email = "test" + new Date().getTime() + "@restmail.net";
-        var password = "iliketurtles";
+        var email = 'test' + new Date().getTime() + '@restmail.net';
+        var password = 'iliketurtles';
 
         return respond(client.signUp(email, password), RequestMocks.signUp)
           .then(
@@ -44,8 +44,8 @@ define([
       });
 
       test('#withKeys', function () {
-        var email = "test" + new Date().getTime() + "@restmail.net";
-        var password = "iliketurtles";
+        var email = 'test' + new Date().getTime() + '@restmail.net';
+        var password = 'iliketurtles';
         var opts = {
           keys: true
         };
@@ -63,9 +63,9 @@ define([
       });
 
       test('#create account with service, redirectTo, and resume', function () {
-        var user = "test" + new Date().getTime();
-        var email = user + "@restmail.net";
-        var password = "iliketurtles";
+        var user = 'test' + new Date().getTime();
+        var email = user + '@restmail.net';
+        var password = 'iliketurtles';
         var opts = {
           service: 'sync',
           redirectTo: 'https://sync.firefox.com/after_reset',
@@ -95,9 +95,9 @@ define([
       });
 
       test('#withService', function () {
-        var user = "test" + new Date().getTime();
-        var email = user + "@restmail.net";
-        var password = "iliketurtles";
+        var user = 'test' + new Date().getTime();
+        var email = user + '@restmail.net';
+        var password = 'iliketurtles';
         var opts = {
           service: 'sync'
         };
@@ -120,9 +120,9 @@ define([
       });
 
       test('#withRedirectTo', function () {
-        var user = "test" + new Date().getTime();
-        var email = user + "@restmail.net";
-        var password = "iliketurtles";
+        var user = 'test' + new Date().getTime();
+        var email = user + '@restmail.net';
+        var password = 'iliketurtles';
         var opts = {
           redirectTo: 'http://sync.firefox.com/after_reset'
         };
@@ -146,9 +146,9 @@ define([
       });
 
       test('#withResume', function () {
-        var user = "test" + new Date().getTime();
-        var email = user + "@restmail.net";
-        var password = "iliketurtles";
+        var user = 'test' + new Date().getTime();
+        var email = user + '@restmail.net';
+        var password = 'iliketurtles';
         var opts = {
           resume: 'resumejwt'
         };
@@ -172,8 +172,8 @@ define([
       });
 
       test('#preVerified', function () {
-        var email = "test" + new Date().getTime() + "@restmail.net";
-        var password = "iliketurtles";
+        var email = 'test' + new Date().getTime() + '@restmail.net';
+        var password = 'iliketurtles';
         var opts = {
           preVerified: true
         };
@@ -193,10 +193,10 @@ define([
       if (! setupEnv.useRemoteServer) {
 
         test('#preVerifyToken', function () {
-          var email = "test" + new Date().getTime() + "@restmail.net";
-          var password = "iliketurtles";
+          var email = 'test' + new Date().getTime() + '@restmail.net';
+          var password = 'iliketurtles';
           var opts = {
-            preVerifyToken: "somebiglongtoken"
+            preVerifyToken: 'somebiglongtoken'
           };
 
           return respond(client.signUp(email, password, opts), RequestMocks.signUp)

--- a/tests/lib/unbundle.js
+++ b/tests/lib/unbundle.js
@@ -33,7 +33,7 @@ define([
                 wrapKB: sjcl.codec.hex.fromBits(sjcl.bitArray.bitSlice(keyAWrapB, 8 * 32, 8 * 64))
               };
 
-              return keyObj
+              return keyObj;
             }
           ).then(
             function(result) {
@@ -53,8 +53,8 @@ define([
               assert.equal(result.kA, '939282904b808c6003ea31aeb14bc766d2ab70ba7dcaa54f820efcf4762b9619', '== kA equal');
               assert.equal(result.wrapKB, '849ac9f71643ace46dcdd384633ec1bffe565852806ee2f859c3eba7fafeafec', '== wrapKB equal');
             }
-          )
-      })
+          );
+      });
 
     });
   }

--- a/tests/lib/verifyCode.js
+++ b/tests/lib/verifyCode.js
@@ -10,21 +10,17 @@ define([
 
   with (tdd) {
     suite('verifyCode', function () {
-      var accountHelper;
       var respond;
       var mail;
       var client;
       var RequestMocks;
-      var ErrorMocks;
 
       beforeEach(function () {
         var env = new Environment();
-        accountHelper = env.accountHelper;
         respond = env.respond;
         mail = env.mail;
         client = env.client;
         RequestMocks = env.RequestMocks;
-        ErrorMocks = env.ErrorMocks;
       });
 
       test('#verifyEmail', function () {
@@ -36,13 +32,13 @@ define([
         return respond(client.signUp(email, password), RequestMocks.signUp)
           .then(function (result) {
             uid = result.uid;
-            assert.ok(uid, "uid is returned");
+            assert.ok(uid, 'uid is returned');
 
             return respond(mail.wait(user), RequestMocks.mail);
           })
           .then(function (emails) {
             var code = emails[0].html.match(/code=([A-Za-z0-9]+)/)[1];
-            assert.ok(code, "code is returned");
+            assert.ok(code, 'code is returned');
 
             return respond(client.verifyCode(uid, code), RequestMocks.verifyCode);
           })
@@ -51,7 +47,7 @@ define([
               assert.ok(result);
             },
             assert.notOk
-          )
+          );
       });
 
       test('#verifyEmailCheckStatus', function () {
@@ -64,25 +60,25 @@ define([
         return respond(client.signUp(email, password), RequestMocks.signUp)
           .then(function (result) {
             uid = result.uid;
-            assert.ok(uid, "uid is returned");
+            assert.ok(uid, 'uid is returned');
 
             return respond(client.signIn(email, password), RequestMocks.signIn);
           })
           .then(function (result) {
-            assert.ok(result.sessionToken, "sessionToken is returned");
+            assert.ok(result.sessionToken, 'sessionToken is returned');
             sessionToken = result.sessionToken;
 
             return respond(client.recoveryEmailStatus(sessionToken),
                     RequestMocks.recoveryEmailUnverified);
           })
           .then(function (result) {
-            assert.equal(result.verified, false, "Email should not be verified.");
+            assert.equal(result.verified, false, 'Email should not be verified.');
 
             return respond(mail.wait(user), RequestMocks.mail);
           })
           .then(function (emails) {
             var code = emails[0].html.match(/code=([A-Za-z0-9]+)/)[1];
-            assert.ok(code, "code is returned: " + code);
+            assert.ok(code, 'code is returned: ' + code);
 
             return respond(client.verifyCode(uid, code),
                     RequestMocks.verifyCode);
@@ -94,10 +90,10 @@ define([
           })
           .then(
             function (result) {
-              assert.equal(result.verified, true, "Email should be verified.");
+              assert.equal(result.verified, true, 'Email should be verified.');
             },
             assert.notOk
-          )
+          );
       });
     });
   }

--- a/tests/mocks/request.js
+++ b/tests/mocks/request.js
@@ -2,6 +2,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+//jscs:disable maximumLineLength
 define(['client/lib/errors'], function (ERRORS) {
   return {
     signUp: {

